app/js/lib/foxbox/foxbox.js

```diff
@@ -180,32 +149,29 @@ export default class Foxbox extends Service {
         // we have no record of previous registrations, we schedule
         // a retry.
         if (!this[p.boxes].length &&
-            !this[p.settings]._localOrigin &&
-            !this[p.settings]._tunnelOrigin) {
+            !this[p.settings].localOrigin &&
+            !this[p.settings].tunnelOrigin) {
           throw new Error('Registration service did not return any boxes.');
         }
 
         if (!this[p.settings].configured) {
           this.selectBox();
         }
-
-        this._dispatchEvent('box-online', true);
```

Comment:
I think we need an event to inform the view that a new box was detected or a previous box was disconnected.
Something like `boxes` (on the same pattern than `services` events). We can then make sure that the view is always in sync with whatever the registration server returns.